apollo-federation/src/operation/mod.rs

```diff
@@ -1537,6 +1553,22 @@ impl FragmentSpreadSelection {
         }
         self.selection_set.any_element(predicate)
     }
+
+    pub(crate) fn for_each_element(
+        &self,
+        parent_type_position: CompositeTypeDefinitionPosition,
+        callback: &mut impl FnMut(OpPathElement) -> Result<(), FederationError>,
+    ) -> Result<(), FederationError> {
+        let inline_fragment = InlineFragment::new(InlineFragmentData {
```

Comment:
guessing this is because `OpPathElement` cannot be fragment spread?

Reply:
It's more because it's what the JS codebase currently does (which I'm guessing is a reflection of the caller's expectations/requirements).